cargo-concordium/src/build.rs

```diff
@@ -659,6 +660,71 @@ pub fn write_json_schema_to_file_v3(
     write_schema_json(path_of_out, contract_name, contract_counter, schema_json)
 }
 
+/// Build the smart contract module and run all integration tests.
+///
+/// The module is built with [`build_contract`] and is placed at
+/// `./concordium-test-out/module.wasm.v1`.
+///
+/// The method discovers all test targets and run them.
+pub fn build_and_run_integration_tests() -> anyhow::Result<()> {
+    eprintln!(
+        "\n{}",
+        Color::Green.bold().paint("Running integration tests ...")
+    );
+
+    let metadata = MetadataCommand::new()
+        .exec()
+        .context("Could not access cargo metadata.")?;
+
+    let mut cargo_args = vec!["test"];
+    let test_args = metadata
+        .root_package()
+        .context("Could not determine package.")?
+        .targets
+        .iter()
+        .filter_map(|t| {
+            if t.kind == ["test"] {
+                Some(["--test", &t.name])
+            } else {
+                None
+            }
+        })
+        .flatten();
+    cargo_args.extend(test_args);
+
+    // TODO: Add option for user to embed schema (and also generate the json
+    // schema)?
+    build_contract(
+        WasmVersion::V1,
+        SchemaBuildOptions::DoNotBuild,
+        Some("concordium-test-out/module.wasm.v1".into()),
+        &[],
+    )?;
```

Comment:
I think the reasonable behaviour would be the following
- support the same options as `build` does.
- if the user does not supply the output path then build it in the usual location in `target/...` (this is also the behaviour of build)

This way we don't pollute the user's root directory unless they tell us so.